src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerService.kt

```diff
@@ -76,9 +77,33 @@ class VisitSchedulerService(
       available = filterNonAssociationByPrisonerId(available, prisonerId)
     }
 
+    populateBookedCounts(available)
+
     return available.sortedWith(compareBy { it.startTimestamp })
   }
 
+  private fun populateBookedCounts(available: List<VisitSession>) {
+    available.forEach {
+      it.openVisitBookedCount = visitBookedCount(it, VisitRestriction.OPEN)
+      it.closedVisitBookedCount = visitBookedCount(it, VisitRestriction.CLOSED)
+    }
+  }
+
+  fun visitBookedCount(session: VisitSession, restriction: VisitRestriction): Int {
+    return visitRepository.findAll(
+      VisitSpecification(
```

Comment:
is that expensive? is there a way of doing this in the JPQL?

Reply:
Yeah could do a query and count.  Its using the current design again which we want to re-visit.

Reply:
This has a backlog item